app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,28 @@
+class CommentsController < ApplicationController
+  before_action :require_login
+
+  def create
+    @comment = Comment.new(comment_params)
+
+    if @comment.save
+      flash[:notice] = "Your comment was saved!"
+      redirect_to trail_path(@comment.trail_id)
+    else
+      flash[:notice] = "Comment was not saved!"
+      redirect_to trail_path(@comment.trail_id)
+    end
+  end
+
+  protected
+  def require_login
+    unless current_user
+      flash[:notice] = "You must be logged in to comment!"
+      redirect_to root_path
+    end
+  end
+
+  private
+  def comment_params
+    params.require(:comment).permit(:description, :trail_id).merge(user_id: current_user.id)
+  end
+end
```

Comment:
A note on `private` vs. `protected` methods. 

* protected methods can be called by any instance of the defining class or its subclasses.

* private methods can be called only from within the calling object. You cannot access another instance's private methods directly.

[source](http://stackoverflow.com/a/3534581/2675670)